### PR TITLE
feat(edit-ema): Expose the languages that a page is available on via REST #27185

### LINKED ISSUE #27185: Edit Page: Expose the languages that a page is available on via REST
### Parent Issue

_No response_

### User Story

As a dotCMS User, I want to be able to quickly determine what languages an HTML Page is available on, so that I can help Content Authors create a page in a language that doesn't exist yet.

### Acceptance Criteria

The new method will be called like this:
```
http://localhost:8080/api/v1/page/{PAGE_ID}/languages
```
And the expected JSON response must look like this:
```json
{
    "entity": [
        {
            "country": "United States",
            "countryCode": "US",
            "defaultLanguage": true,
            "id": 1,
            "isoCode": "en-us",
            "language": "English",
            "languageCode": "en"
            "translated": true // The page HAS an English version
        },
        {
            "country": "Espana",
            "countryCode": "ES",
            "defaultLanguage": false,
            "id": 2,
            "isoCode": "es-es",
            "language": "Espanol",
            "languageCode": "es",
            "translated": false // The page DOES NOT have a Spanish version
        }
    ],
    "errors": [],
    "i18nMessagesMap": {},
    "messages": [],
    "pagination": null,
    "permissions": []
}
```

### Steps to Reproduce

1. Create three languages in your dotCMS instance:
1.1. English
1.2. Spanish
1.3. French
2. Create an HTML Page **ONLY in English**. Copy its Identifier.
3. Call the new REST method: http://localhost:8080/api/v1/page/{PAGE_ID}/languages and pass down the page's Identifier.
4. The JSON response must look like the one in the **Acceptance Criteria** section. The `translated` must be present for ALL languages, but its value must be `true` for the English language only.

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### Proposed Changes
* Exposes a new method that allows Users to get the list of available languages in dotCMS and, based on the provided page Identifier, adds a flag attribute for each language that indicates whether the page has a version in such a language or not.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java [hunk 1]
```diff
@@ -1137,4 +1136,45 @@ public ResponseEntityPageWorkflowActionsView findAvailableActions(@Context final
...
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response checkPageLanguageVersions(@Context final HttpServletRequest request,
+                                              @Context final HttpServletResponse response,
+                                              @PathParam("pageId") final String pageId) {
+        final User user = new WebResource.InitBuilder(webResource).requestAndResponse(request, response)
+                .rejectWhenNoUser(true)
+                .requiredBackendUser(true)
+                .init().getUser();
+        Logger.debug(this, String.format("Check the languages that page '%s' is available on", pageId));
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:1168]: use supplier

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java [hunk 2]
```diff
@@ -1137,4 +1136,45 @@ public ResponseEntityPageWorkflowActionsView findAvailableActions(@Context final
...
+        try {
+            final List<ExistingLanguagesForPageView> languagesForPage =
+                    this.pageResourceHelper.getExistingLanguagesForPage(pageId, user);
+            return Response.ok(new ResponseEntityView<>(languagesForPage)).build();
+        } catch (final Exception e) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:1173]: do not need to catch, jersey took and map the exception

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java (hunk 2 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java]: Done!

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java (hunk 1 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java]: Done!

**Review by freddyDOTCMS — APPROVED** [review]

**Pull request merged**